app/Models/GeneralAssemblies/GeneralAssembly.php

```diff
@@ -133,11 +155,38 @@ public function close(): void
         if (!$this->isOpen()) {
             throw new \Exception("tried to close general assembly when it was not open");
         }
-        foreach ($this->questions()->get() as $question) {
+        foreach ($this->presenceChecks()->get() as $question) {
             if ($question->isOpen()) {
                 $question->close();
             }
         }
-        $this->update(['closed_at'=>now()]);
+        $this->update(['closed_at' => now()]);
+    }
+
+    /**
+     * @return int The number of presence checks needed for a user to be counted as an attended user.
+     */
+    public function getPresenceChecksNeededAttribute(): int
+    {
+        $presenceCheckCount = $this->presenceChecks()->count();
+        return $presenceCheckCount <= 2 ? $presenceCheckCount : $presenceCheckCount - 2;
+    }
```

Comment:
Why don't you use https://laravel.com/docs/10.x/eloquent-mutators#defining-an-accessor ?

Reply:
I've never used it before 😅